### ISSUE TITLE
About Battle Prediction, Damage Control, and Sunk!

### ISSUE THREAD

**Issue by ReiFan49** [dragonjet/KC3Kai#493]
ok, let's start a darker topic.

### scroll down if it's hard to understand (sorry)
maybe i should say this as feature request, because damage control/goddess can prevent any ship from sinking on that battle.
as we know, damage control cancels the damage and heals the kanmusu about 20% of the HP

**Start Battle with Taiha Condition**
![image](https://cloud.githubusercontent.com/assets/9847780/9109675/99ff8b38-3c61-11e5-93f3-a7aa333fa65e.png)
**Kuso**
![image](https://cloud.githubusercontent.com/assets/9847780/9109681/abc7c22c-3c61-11e5-9d28-8827def65a71.png)
**After Damage Control Consumed**
![image](https://cloud.githubusercontent.com/assets/9847780/9109701/d505eab0-3c61-11e5-949c-26d1033e4909.png)
*these pics are taken during last E-4 Final Kill, no API data recorded and only videos exists to represent it*

nah at the moment, i'm gonna put some stamp on the kanmusu which is sunk or knocked out during PvP (PvP knockout = sunk in real battle)
however, before implementing that, i need to make sure the kanmusu that started battle with **heavy-damaged state did not bring a damage control**, if the kanmusu brought a damage control during the heavily-damaged state, as i said, the HP recovers back to 20% (sad that, from 2 attempts, **all of my experiment kanmusu have a zero remainder of 5 of its maximum HP, so i cannot ensure whether the devs rounded it down or not**), and further calculation should be treated as normal state of kanmusu (which is cannot sunk).

previously, i want to check a knocked out kanmusu on PvP using such technique, however fails ._.
![this](https://fbcdn-sphotos-d-a.akamaihd.net/hphotos-ak-xaf1/v/t1.0-9/11822736_10204647464533150_5011702513831341999_n.jpg?oh=9306ed8cb8f33e685c5a74845de664a3&oe=5653E370&__gda__=1448317380_6cf3d7fbde6133d9b12f122d243a6446)
*just remember, 1HP on screen is different on API data*

### **tl;dr ver**
what i mean in here,

battle prediction/calculation might break if there's a damage control (or similar) equipped to the respecting kanmusu that starts the battle with badly damaged condition (as stated above).

(PS: poor ryuujou, i'm sorry for such things)

**Comment by bamboo3250** [comment]
You have a point. There are so many factors to consider in the battle and we cannot force any KC3Kai developer to try all of them. In particular, in this case, we might need help from other ttks to collect API data for us. 

@dragonjet can we include the response data of each battle in the Strategy Room so that other players can find it easily? We can ask their help by that way.

**Comment by dragonjet** [comment]
sure, we can ship it aug 9, tho too close to event but better than after xD

**Comment by ReiFan49** [comment]
PvP hp bar? oh lol i just done the similar thing i guess on my fix,
because i focused on my thing before merging it with the main one

(PS: the pic is taken about several days ago)

**Comment by c933103** [comment]
off-topic: do girl sinking in pvp consume damecon if they equip it?

btw, do kc3 kai currently work well with fleet command facility?

**Comment by ReiFan49** [comment]
FCF? oh damn, since i didn't have it, i cannot test it (the escorted kanmusu right?)

also, it's not off-topic, because i've done it right above your post lol. (Ryuujou attacked by Yamato at the end of 2nd shelling phase, from 7HP to **-261HP** with damecon indeed, 0 exp)

**Comment by dragonjet** [comment]
wow you risked a damecon for it xD

hm going back to the main issue, so as I understand, we'll need an update to the Battle Calculation script by @Javran, that when a ship `ending HP <= 0`, and when she has damecon, then `hp = max / 5`?

**Comment by nicky008** [comment]
And we have two kinds of damecon in the game, normal and goddess.

* `hp = max / 5` for the normal.
* `hp = max` for the goddess.

Need to consider this too...

**Comment by bamboo3250** [comment]
And we also can have multiple Damecon (up to 4) for a ship, which can be a mix of normal and goddess.

**Comment by dragonjet** [comment]
gg

**Comment by ethrundr** [comment]
@bamboo3250 so... you gonna have your ship sortie with 4 damecons? :smile: 
gonna wait to watch them hitting abyssal with hammers XD

**Comment by bamboo3250** [comment]
@ethrundr of course not, I haven't used any Damecon up until now. But that's possible for someone in the deepest of desperation during the coming event :laughing:

**Comment by ethrundr** [comment]
@bamboo3250 ah, true. events can make the calmest sheep into most ferocious wolf in an instant :laughing:

**Comment by dragonjet** [comment]
by the way I have *Ship Condition Stamps* enabled on settings and seems to not work on natsuiro, though I made sure last time I included the `addClass()` and the CSS for it.

**Comment by ReiFan49** [comment]
css play requires `position:relative` on its parent and `position:absolute` on the stamp itself to make it appears above the kanmusu name

**Comment by Javran** [comment]
@dragonjet what's the status of ship condition stamps on Natsuiro?
I just checked out the latest repo, but it doesn't appear, the "ship condition stamps" option is already checked.

**Comment by dragonjet** [comment]
it should work, since I also got it on a normal sortie

![ss 2015-08-21 at 04 14 31](https://cloud.githubusercontent.com/assets/557312/9431685/4b41c8da-4a49-11e5-96d9-135af3768333.jpg)

**Comment by Javran** [comment]
@dragonjet oh sorry I didn't mean that. I thought ship condition stamps were those expedition circles.. what about them?

**Comment by dragonjet** [comment]
ah are we still adding those icons? iirc you have a plan for the panel itself on #685 if we do implement it, there will be no need for the expedition balls

though I'd like to make a separate tracker summary for expedition-related features. a list, something like #582 and ##633. right now I'm not sure what's our latest plan for both panel and strategy room

**Comment by dragonjet** [comment]
created #722 for expedition features

**Comment by Javran** [comment]
@dragonjet I see, just wondering if they are actually present somewhere in natsuiro, if the purposed plan is implemented, there is indeed no need for them anymore.

**Comment by dragonjet** [comment]
ah right now, they're not added to natsuiro

**Comment by ReiFan49** [comment]
![image](https://cloud.githubusercontent.com/assets/9847780/9526934/41e73cf0-4d16-11e5-82a4-b59184223844.png)
now i need to ask the admiral for the battle data... if possible (or i'll try to gather by go to the event again..... ps: not in the mood lol)

**Comment by ReiFan49** [comment]
https://github.com/andanteyk/ElectronicObserver/blob/master/ElectronicObserver/Other/Information/kcmemo.md
there are many important things written here
